From a671b535dcef24d0bb4329f5dde51be742715199 Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Tue, 17 Apr 2018 13:57:21 +0800
Subject: [PATCH 3611/5242] MLK-18042-01 driver: pwm: fix pwm pre-scale div
 config

commit  b90224e2bd9dabe7c754872c9c891055dbb9b08c from
https://source.codeaurora.org/external/imx/linux-imx.git

In i.MX7ULP TPM PWM module, it has a pre-scale divider,
this divider setting is missed, so fix it.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Reviewed-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pwm/pwm-tpm.c |   10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/pwm/pwm-tpm.c b/drivers/pwm/pwm-tpm.c
index 32a7116..0434d2a 100644
--- a/drivers/pwm/pwm-tpm.c
+++ b/drivers/pwm/pwm-tpm.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2017 NXP.
+ * Copyright 2017-2018 NXP.
  *
  * The code contained herein is licensed under the GNU General Public
  * License. You may obtain a copy of the GNU General Public License
@@ -54,7 +54,7 @@ static int tpm_pwm_config(struct pwm_chip *chip, struct pwm_device *pwm,
 			  int duty_ns, int period_ns)
 {
 	struct tpm_pwm_chip *tpm = to_tpm_pwm_chip(chip);
-	int ret, div = 0;
+	int ret, val, div = 0;
 	unsigned int period_cycles, duty_cycles;
 	unsigned long rate;
 	u64 c;
@@ -79,6 +79,12 @@ static int tpm_pwm_config(struct pwm_chip *chip, struct pwm_device *pwm,
 			return ret;
 	}
 
+	/* set the pre-scale */
+	val = readl(tpm->base + TPM_SC);
+	val &= ~0x7;
+	val |= div;
+	writel(val, tpm->base + TPM_SC);
+
 	period_cycles = c;
 	c *= duty_ns;
 	do_div(c, period_ns);
-- 
1.7.9.5

